### PR TITLE
PP-418: allow punctuation in names

### LINKED ISSUE #2580: PP-418 OIP fout bij registratie
Wanneer een inwoner voor het eerst inlogt op [mijn.zwolle.nl](http://mijn.zwolle.nl/) of geen profiel bezit en de gebruiker heeft een “.” in de naam zitten die zo is geregistreerd binnen de BRP verschijnt er een foutmelding dat dit teken niet wordt geaccepteerd. Omdat deze zelf niet aangepast kan worden door de gebruiker en het wel voorkomt in de BRP zou er geen beperking moeten zijn op het “.” teken. 

Op productie is dit voorgekomen bij deze persoon:
Image

**Acties ondernomen:**
Getest met BSN 999991000 op de acceptatieomgeving, deze staat ook in de testset van Haalcentraal. In deze naam: Liselotte-Anne Daniëlle Celèste Elise 26 Avril Stéphane Maroni jr. Isiah dé Maria de las Mercedes Rosalía J'adore Grace a Dieu Rhiannon San-che-he-ray Lou'Lou Elona L'vovna d'Alessandra Ariëlle zit ook een “.” en hierbij wordt ook de foutmelding gegeven:
Image

**Welke impact heeft dit probleem op de bedrijfsvoering?**
Gebruikers die deze foutmelding krijgen kunnen niet registreren en kunnen geen zaken volgen in het portaal. 

**Gerelateerde issues:**
https://taiga.maykinmedia.nl/project/open-inwoner/issue/1503

De aanname was toen gedaan dat de namen in de BRP alleen de genoemde tekens zou bevatten, maar dat is onterecht: 
[Blaming open-inwoner/src/open_inwoner/utils/validators.py at f6ea774435819ac82b4d4e2de72d30ed050e7dfc · maykinmedia/open-inwoner](https://github.com/maykinmedia/open-inwoner/blame/f6ea774435819ac82b4d4e2de72d30ed050e7dfc/src/open_inwoner/utils/validators.py#L77) 

De BRP hanteert een andere reguliere expressie van de toegestane tekens hiervoor, met diacrieten en punten in de naam: [a-zA-Z0-9À-ž \.\-\']

Dit dient gecorrigeerd te worden in de applicatie:
Image

### PR DESCRIPTION
**Pull request by swrichards**
Validation of first_name, infix, and last_name is moved from the user model to the form layer, which allows updates from the BRP with names containing punctuation and diacritics. This is a temporary solution until a general approach to name validation has been agreed upon.

### CODE REVIEW

File: src/open_inwoner/utils/validators.py [hunk 1]
```diff
@@ -80,13 +80,22 @@ def _check_for_invalid_chars(self, value: str) -> None:
...
-    regex = r"^[\w'’\- ]+\Z"
+    regex = r"^[\w’’\- ]+\Z"
     message = _(
         "Please make sure your input contains only valid characters "
         "(letters, numbers, apostrophe, dash, space)."
     )
 
 
+@deconstructible
+class NameFieldValidator(RegexValidator):
```
Comment (swrichards) [inline src/open_inwoner/utils/validators.py]: @pi-sigma I think it would be bad having a unit test for this, with some examples of edge case names taken from: https://github.com/maykinmedia/ansible-maykin-infra/blob/main/roles/haalcentraal_mock/files/personen-mock/personen.json

**Comment by pi-sigma** [comment]
Instead of adding punctuation to our allowlist, we considered disabling validation of names altogether because it inevitably excludes legitimate names in an arbitrary fashion. Even the BRP list of allowed names excludes names with Greek diacritics, Arabic names, and many others. From a security perspective, Django itself goes a long way towards addressing the most pressing issues. (ORM against SQL injection, auto-escaping in templates against XSS). We might still want to exclude certain names; cf. the discussion in this article: [https://www.b-list.org/weblog/2018/feb/11/usernames/](https://www.b-list.org/weblog/2018/feb/11/usernames/).

**Review by pi-sigma — APPROVED** [review]
